### DISCUSSION TITLE
Additional help needed: WIP Ghosthunter (SLUS-20993)

### DISCUSSION THREAD

**Discussion by hoyiliang** [garungorp/MouseInjectorDolphinDuck#21] [Show and tell]
Referring to the tutorial created by @garungorp :
[https://www.youtube.com/watch?v=LLtaIrMyQBc&ab_channel=garungorp](url)

I have successfully developed the injection codes until the **isPaused flag search and programming.**

I am currently stuck in the **camBase sanity check** section where the tutorial tells me to screenshot the memory region of different save states. 

This is because this game has **3 camera modes** (which I have mapped them), and they shares a **single camBase** (and they are kinda far apart with each other).

I do not know exactly where to screenshot. I hope someone is able to progress past this section I'm stuck with and share here. Thanks!

Tips: if you are unable to get the addresses like in the YouTube tutorial, please check in your Cheat Engine settings: _Edit -> Settings -> Scan Settings_ and make sure **MEM_MAPPED** is checked (**by default it is not**), then you will be able to find the addresses you need. 

### Software Information:
PCSX2 Nightly - v1.7.4974
Cheat Engine 7.5
Windows 11 Pro
IDE: VS CODE (with some extensions for C development)

### My current WIP files:
[games.zip](https://github.com/user-attachments/files/16942297/games.zip)

This zip contains 3 files:
1. game.c which added new entries for the game.
2. ps2_ghosthunter.c which contains my current development progress for the mouse injection on this game.
3. GHOSTHUNTER.CT which contains the required offsets/ranges/pointers/addresses while I am developing this.

Thanks again for @garungorp to create the amazing tutorial to make the development of this possible!

**Comment by hoyiliang** [comment]
Currently you can already play the game with mouse injection with the current state but there are no cambase sanity checks in any way.

**Reply by Clippy95** [reply]
Seems like the Sanity check is just an extra check to be sure that your camera_base is correct, I've done Mousehook for some X360 games and haven't had a need for it, it's a smart idea to check for it though, from what I understand you read the address of your camera_base which will give you the camera_base, you read the memory region of camera_base in different states and compare those states and find any similarities and you calculate offsets accordingly, what's the exact problem are you having with it?
